src/Utils/citesupport.ts

```diff
@@ -0,0 +1,453 @@
+"use strict";
+import CiteWorker from "./cite.worker.ts";
+/* global Word OfficeExtension*/
+
+class CiteSupport {
+  config: {
+    debug: boolean;
+    mode: string;
+    defaultLocale: string;
+    defaultStyle: string;
+    citationIdToPos: {};
+    citationByIndex: object[];
+    processorReady: boolean;
+    citationData: object[];
+  };
+  worker: Worker;
+  constructor(citationData: object[]) {
+    this.config = {
+      debug: true,
+      mode: "in-text",
+      defaultLocale: "en-US",
+      defaultStyle: "american-sociological-association",
+      citationIdToPos: {},
+      citationByIndex: [],
+      processorReady: false,
+      citationData: citationData,
+    };
+    var me = this;
+    this.worker = new CiteWorker();
+    this.worker.onmessage = function (e) {
```

Comment:
Is it not easier to make this an arrow function instead of using the `this -> me` hack?